### PR TITLE
cmd/systray: rebuild menu on pref change, assorted other fixes

### PR DESCRIPTION
**Pull request by willnorris**
- rebuild menu when prefs change outside of systray, such as setting an exit node
- refactor onClick handler code
- compare lowercase country name, the same as macOS and Windows (now sorts Ukraine before USA)
- fix "connected / disconnected" menu items on stopped status
- prevent nil pointer on "This Device" menu item

Updates #1708

Change-Id: Ia101a4a3005adb9118051b3416f5a64a4a45987d

### CODE REVIEW

File: cmd/systray/systray.go [hunk 1]
```diff
@@ -36,6 +36,8 @@ var (
 	localClient tailscale.LocalClient
 	chState     chan ipn.State // tailscale state changes
 
+	chRebuild chan int // triggers a menu rebuild
```
Comment (bradfitz) [inline cmd/systray/systray.go:39]: chan struct{} maybe so you don't need to worry the reader about what the int or bool or whatever means?

File: cmd/systray/systray.go [hunk 2]
```diff
@@ -337,14 +339,27 @@ func (menu *Menu) eventLoop(ctx context.Context) {
...
-			menu.rebuild(fetchState(ctx))
 
 		case <-menu.quit.ClickedCh:
 			systray.Quit()
 		}
 	}
 }
 
+// onClick registers a click handler for a menu item.
+func onClick(ctx context.Context, item *systray.MenuItem, fn func()) {
```
Comment (bradfitz) [inline cmd/systray/systray.go:350]: should fn take the context too? you have it.

Then the funcs can select on ctx.Done themselves and not block forever on shutdown

File: cmd/systray/systray.go [hunk 3]
```diff
@@ -425,25 +443,14 @@ func (menu *Menu) rebuildExitNodeMenu(ctx context.Context) {
...
-					case <-ctx.Done():
-						return
-					case menu.exitNodeCh <- nodeID:
-					}
-				}
-			}
-		}()
+	setExitNodeOnClick := func(item *systray.MenuItem, nodeID tailcfg.StableNodeID) {
+		onClick(ctx, item, func() {
+			menu.exitNodeCh <- nodeID
```
Comment (bradfitz) [inline cmd/systray/systray.go:448]: if you had ctx, you could select on this send + ctx.Done recv?

File: cmd/systray/systray.go [hunk 4]
```diff
@@ -558,7 +565,7 @@ type mullvadPeers struct {
 func (mp mullvadPeers) sortedCountries() []*mvCountry {
 	countries := slices.Collect(maps.Values(mp.countries))
 	slices.SortFunc(countries, func(a, b *mvCountry) int {
-		return cmp.Compare(a.name, b.name)
+		return cmp.Compare(strings.ToLower(a.name), strings.ToLower(b.name))
```
Comment (bradfitz) [inline cmd/systray/systray.go:568]: this is a little painful to see allocs in a sort func

see https://en.wikipedia.org/wiki/Schwartzian_transform

I sent https://github.com/tailscale/tailscale/pull/14458

File: cmd/systray/systray.go [hunk 5]
```diff
@@ -574,7 +581,7 @@ type mvCountry struct {
 func (mc *mvCountry) sortedCities() []*mvCity {
 	cities := slices.Collect(maps.Values(mc.cities))
 	slices.SortFunc(cities, func(a, b *mvCity) int {
-		return cmp.Compare(a.name, b.name)
+		return cmp.Compare(strings.ToLower(a.name), strings.ToLower(b.name))
```
Comment (bradfitz) [inline cmd/systray/systray.go:584]: same

**Review by bradfitz — APPROVED** [review]

File: cmd/systray/systray.go (hunk 1 above)
Comment (willnorris) [reply cmd/systray/systray.go]: done

File: cmd/systray/systray.go (hunk 2 above)
Comment (willnorris) [reply cmd/systray/systray.go]: done

File: cmd/systray/systray.go (hunk 3 above)
Comment (willnorris) [reply cmd/systray/systray.go]: done

File: cmd/systray/systray.go (hunk 4 above)
Comment (willnorris) [reply cmd/systray/systray.go]: fixed

File: cmd/systray/systray.go (hunk 5 above)
Comment (willnorris) [reply cmd/systray/systray.go]: done

**Pull request merged**